db/attachment_compaction.go

```diff
@@ -211,15 +211,11 @@ func getAttachmentSyncData(dataType uint8, data []byte) (*AttachmentCompactionDa
 	var documentBody []byte
 
 	if dataType&base.MemcachedDataTypeXattr != 0 {
-		body, xattr, _, err := parseXattrStreamData(base.SyncXattrName, "", data)
+		body, xattrs, err := sgbucket.DecodeValueWithXattrs([]string{base.SyncXattrName}, data)
 		if err != nil {
-			if errors.Is(err, base.ErrXattrNotFound) || errors.Is(err, base.ErrXattrInvalidLen) {
-				return nil, nil
-			}
-			return nil, err
+			return nil, fmt.Errorf("Could not parse DCP attachment sync data: %w", err)
```

Comment:
This looks like a functional change if the xattr isn't found.  Should it be checking if value was found?

Reply:
This looks like a functional change but `parseXattrStreamData` never returned `ErrXattrNotFound`. That said, I should catch `ErrXattrInvalidLen` though realistically this is also an error condition where we probably could consider stopping.